ozark/src/main/java/org/glassfish/ozark/jersey/OzarkFeature.java

```diff
@@ -87,27 +101,47 @@ public void configure(FeatureContext context) {
         if (config.isRegistered(ViewResponseFilter.class)) {
             return;     // already registered!
         }
-        final boolean enableOzark = config.getClasses().stream().anyMatch(this::isController)
-                || config.getInstances().stream().map(o -> o.getClass()).anyMatch(this::isController);
-        if (enableOzark) {
-            context.register(ViewRequestFilter.class);
-            context.register(ViewResponseFilter.class);
-            context.register(ViewableWriter.class);
-            context.register(BindingInterceptorImpl.class);
-            context.register(OzarkModelProcessor.class);
-            context.register(CsrfValidateInterceptor.class);
-            context.register(CsrfProtectFilter.class);
-
-            // Initialize application config object in Mvc class
-            final BeanManager bm = CDI.current().getBeanManager();
-            final MvcContextImpl mvc = (MvcContextImpl) newBean(bm, MvcContextImpl.class);
-            mvc.setConfig(config);
-            mvc.setContextPath(servletContext.getContextPath());
+        Set<Class<?>> controllers = Stream.concat(config.getClasses().stream(), config.getInstances().stream().map(o -> o.getClass()))
+                .filter(this::isController).collect(Collectors.toSet());
+        if (controllers.isEmpty() || hasUnsupportedTypes(controllers)) {
```

Comment:
To be honest, I'm not completely sure if we should abort the initialization in case of a single problematic controller. Wouldn't it make more sense to just log a warning and continue?